cgo/clapack/genLapack.pl

```diff
@@ -340,6 +373,17 @@ sub processParamToGo {
 
 		my $goType = $typeConv{$type};
 
+		$goType =~ /^\[\]/ && do {
+			my $typeElem = substr $goType, 2;
+			my $bp = << "EOH";
+var _$var *$typeElem
+if len($var) > 0 {
```

Comment:
Looking at the result and rethinking it again I'm not sure anymore if we should do this, and much less for all slice parameters. I triggered this because of Dgehrd and its tau slice parameter that is not needed for small (corner-case) values of n. There we panicked unnecessarily. But on the other hand, for example in Dgehrd `work` must always have length at least 1, its first element is always written to. If work has zero length, we unnecessarily postpone the panic/segfault to inside the C.LAPACKE call.

Dealing with this selectively is too much work and hard to automatize for code generation. It's not ideal but keeping the current status is probably better for the moment? The behavior of the calls via the `clapack` package will not be 100% equivalent to the underlying implementation (for corner-case or incorrect arguments) but it may not matter much since in the vast majority of cases it will be us doing these calls anyway from the `cgo` package.

Of course, this does not include the switch to `_work` calls, I do support doing that.